### ISSUE TITLE
[V3] Config / driver singletons

### ISSUE THREAD

**Issue by mikeshardmind** [Cog-Creators/Red-DiscordBot#1511]
### Type:

- [ ] Suggestion
- [X] Bug

### Brief description of the problem

Driver isn't a singleton, new instances fail to write properly

### Actual behavior

I'm not 100% sure on the internals, but I'll include enough steps below that this can be tracked back quickly. This is also a newer issue somewhat

### Steps to reproduce
`````
[p]eval
```py
from redbot.core.config import Config
from redbot.core import modlog

# how modlog._conf is defined:
_modlog_type = type("ModLog", (object,), {})
x = Config.get_conf(_modlog_type, 1354799444)

y = modlog._conf

assert x.driver == y.driver
```
`````

**Comment by tekulvw** [comment]
This is actually an issue with handling the fake type object. If the `cog_name` kwarg is used config acts as expected.

**Comment by tekulvw** [comment]
Fixed with #1518
